### DISCUSSION TITLE
ITMS-90338: Non-public API usage

### DISCUSSION THREAD

**Discussion by arjunkr92** [dotnet/maui#17422] [Q&A] [answered]
MAUI iOS App got rejected because of following error:

> ITMS-90338: Non-public API usage - The app references non-public symbols in : _ubrk_openRules, _ucal_add, _ucal_close, _ucal_get, _ucal_getAttribute, _ucal_getKeywordValuesForLocale, _ucal_getNow, _ucal_getTimeZoneDisplayName, _ucal_getTimeZoneIDForWindowsID, _ucal_getWindowsTimeZoneID, _ucal_open, _ucal_openTimeZoneIDEnumeration, _ucal_set, _ucal_setMillis, _ucol_close, _ucol_closeElements, _ucol_getOffset, _ucol_getRules, _ucol_getSortKey, _ucol_getStrength, _ucol_getVersion, _ucol_next, _ucol_open, _ucol_openElements, _ucol_openRules, _ucol_previous, _ucol_safeClone, _ucol_setAttribute, _ucol_setVariableTop, _ucol_strcoll, _ucurr_forLocale, _ucurr_getName, _udat_close, _udat_countSymbols, _udat_format, _udat_getSymbols, _udat_open, _udat_setCalendar, _udat_toPattern, _udata_setCommonData, _udatpg_close, _udatpg_getBestPattern, _udatpg_open, _uenum_close, _uenum_count, _uenum_next, _uldn_close, _uldn_keyValueDisplayName, _uldn_open, _uloc_canonicalize, _uloc_countAvailable, _uloc_getAvailable, _uloc_getBaseName, _uloc_getCharacterOrientation, _uloc_getCountry, _uloc_getDefault, _uloc_getDisplayCountry, _uloc_getDisplayLanguage, _uloc_getDisplayName, _uloc_getISO3Country, _uloc_getISO3Language, _uloc_getKeywordValue, _uloc_getLCID, _uloc_getLanguage, _uloc_getName, _uloc_getParent, _uloc_setKeywordValue, _ulocdata_getCLDRVersion, _ulocdata_getMeasurementSystem, _unorm2_getNFDInstance, _unorm2_getNFKCInstance, _unorm2_getNFKDInstance, _unorm2_isNormalized, _unum_close, _unum_getAttribute, _unum_getSymbol, _unum_open, _unum_toPattern, _ures_close, _ures_getByKey, _ures_getSize, _ures_getStringByIndex, _ures_open, _usearch_close, _usearch_first, _usearch_getBreakIterator, _usearch_getMatchedLength, _usearch_last, _usearch_openFromCollator, _usearch_setPattern, _usearch_setText.

I'm using Visual Studio 2022 Comunity Edition (v17.7) to compile and build the app. The app targets only iOS platform, I have removed Windows, Android and Mac from the targetted platforms list.

Third-Party Nuget Packages Installed:
- Newtonsoft.Json
- Onion.Maui.Googlemaps
- SkiaSharp.Extended.UI.Maui
- Xamarin.Firebase.iOS.Database
- CommunityToolkit.Maui

Can anyone suggest a solution for this?

**Comment by JohnHDev** [comment]
You might be building your debug configuration and signing that, rather than the release configuration which strips all unused APIs (including all the private APIs).

**Comment by arjunkr92** [comment, marked as answer]
Thank you @JohnHDev for pointing me in the right direction, the culprit was "Xamarin.Firebase.iOS.Database" plugin, it worked fine in Debug builds but in the release build it was failing.

Someone faced the same issue here: https://github.com/xamarin/GoogleApisForiOSComponents/issues/617

So I replaced it with FirebaseDatabase.net plugin.
